finetune/lora.py

```diff
@@ -253,13 +257,22 @@ def train(
 
 @torch.no_grad()
 def validate(
-    fabric: L.Fabric, model: GPT, val_data: List[Dict], tokenizer: Tokenizer, longest_seq_length: int
+    fabric: L.Fabric, 
+    model: GPT, 
+    val_data: List[Dict], 
+    tokenizer: Tokenizer, 
+    longest_seq_length: int,
+    padded_vocab_size: int,
 ) -> torch.Tensor:
     fabric.print("Validating ...")
     model.eval()
     losses = torch.zeros(eval_iters)
     for k in range(eval_iters):
         input_ids, targets = get_batch(fabric, val_data, longest_seq_length)
+        if (input_ids > padded_vocab_size).any():
```

Comment:
I think there was a misunderstanding. My proposal was to only move the error checking there. The reason I wrote down the long explanation is to exactly avoid the confusion between "code is duplicated, therefore disassemble the scripts into a common trainer utility" vs. making the scrips self contained as much as possible. 

If it was between this and the previous version, I would strongly prefer the previous.

Reply:
I see. The problem is that the error checking requires loading batches, and then we can't have the error checking function in utils because each it would then have to import `get_batch` from the individual scripts. So, maybe we should duplicate the error checking code in the individual scripts similar to `get_batch`?